src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing/CSharpSourceGeneratorTest`2.cs

```diff
@@ -27,5 +28,19 @@ protected override CompilationOptions CreateCompilationOptions()
 
         protected override ParseOptions CreateParseOptions()
             => new CSharpParseOptions(DefaultLanguageVersion, DocumentationMode.Diagnose);
+
+        public IncrementalGeneratorExpectedState IncrementalGeneratorState
```

Comment:
💡 I think we can remove this in favor of changing the type of `IncrementalGeneratorStepStates` to a new type derived from `Dictionary<Type, IncrementalGeneratorExpectedState>` that provides a new indexer that creates the value on demand. In practice in means you'd have to type this, but it seems acceptable:

```
IncrementalGeneratorStepStates =
{
    [typeof(TSourceGenerator)] =
    {
        ExpectedStepStates =
        {
            // ...
        },
    },
},
```